src/main/java/io/github/programminglife2016/pl1_2016/parser/database/SimpleDatabase.java

```diff
@@ -0,0 +1,421 @@
+        return fetchLinks(nodes);
+
+    }
+
+    /**
+     * Fetch all positions of segments in nodeCollection in database.
+     * @param  nodeCollection The nodecollection for which the positions should be fetched
+     * @return collection of nodes in database
+     * @throws SQLException thrown if SQL connection or query is not valid
+     */
+    public NodeCollection fetchPositions(NodeCollection nodeCollection) throws SQLException {
+        for (Node node: nodeCollection.values()) {
+            List<Integer> positions = fetchPosition(node.getId());
+            node.setXY(positions.get(0), positions.get(1));
+        }
+        return nodeCollection;
+
+    }
+
+    /**
+     * Fetch position of segment based on id of the segment.
+     *
+     * @param id the id of the segment
+     * @return the positions as List<Integer>
+     */
+    private List<Integer> fetchPosition(int id) {
+        Statement stmt = null;
+        String query = "select positionx, "
+                + "positiony "
+                + "from " + SEGMENT_TABLE
+                + " WHERE segment_id='" + id + "'";
+        List<Integer> res = new ArrayList<>();
+        try {
+            stmt = connection.createStatement();
+            ResultSet rs = stmt.executeQuery(query);
+            if (rs.next()) {
+                int x = rs.getInt("positionx");
+                int y = rs.getInt("positiony");
+                res.addAll(Arrays.asList(x, y));
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return res;
+
+    }
+
+    /**
+     * Update the positions of the nodeCollection into the database
+     * @param nodeCollection the nodecollection with the correct positions
+     * @throws SQLException thrown if SQL connection or query is not valid
+     */
+    public void updatePositions(NodeCollection nodeCollection) throws SQLException {
+        for (Node node : nodeCollection.values()) {
+            updatePosition(node.getId(), node.getX(), node.getY());
+        }
+    }
+    /**
+     * Update the positions of the node whit id id into the database
+     * @param id the id of the node to update
+     * @param x the correct x for the node
+     * @param y the correct y for the node
+     * @throws SQLException thrown if SQL connection or query is not valid
+     */
+    public void updatePosition(int id, int x, int y) throws SQLException {
+        Statement stmt = null;
+        String query = "UPDATE " + SEGMENT_TABLE
+                + " SET positionx = '"
+                + x + "', positiony = '"
+                + y + "' WHERE segment_id = '"
+                + id + "'";
+        try {
+            System.out.println(query);
+            stmt = connection.createStatement();
+            stmt.execute(query);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            if (stmt != null) {
+                stmt.close();
+            }
+        }
+    }
+
+
```

Comment:
Can you start with an capital "f" and end the line with a period. Checkstyle will throw an error here.